packages/node-core/src/indexer/testing.service.ts

```
@@ -0,0 +1,252 @@
+      for (let i = 0; i < test.expectedEntities.length; i++) {
+        const expectedEntity = test.expectedEntities[i];
+        const actualEntity = await store.get(expectedEntity._name, expectedEntity.id);
+        const attributes = actualEntity as unknown as CreationAttributes<Model>;
+        const failedAttributes: string[] = [];
+        let passed = true;
+        Object.keys(attributes).map((attr) => {
+          const expectedAttr = (expectedEntity as Record<string, any>)[attr];
+          const actualAttr = (actualEntity as Record<string, any>)[attr];
+          if (!isEqual(expectedAttr, actualAttr)) {
+            passed = false;
+            failedAttributes.push(
+              `\t\tattribute: "${attr}":\n\t\t\texpected: "${expectedAttr}"\n\t\t\tactual:   "${actualAttr}"`
+            );
+          }
+        });
+
+        if (passed) {
+          logger.info(`Entity check PASSED (Entity ID: ${expectedEntity.id})`);
+          passedTests++;
+        } else {
+          logger.warn(`Entity check FAILED (Entity ID: ${expectedEntity.id})`);
+          this.failedTestsSummary.push({
+            testName: test.name,
+            entityId: expectedEntity.id,
+            entityName: expectedEntity._name,
+            failedAttributes: failedAttributes,
+          });
+
+          failedTests++;
+        }
+      }
+
+      this.totalPassedTests += passedTests;
+      this.totalFailedTests += failedTests;
+
+      logger.info(
+        `Test: ${test.name} completed with ${chalk.green(`${passedTests} passed`)} and ${chalk.red(
+          `${failedTests} failed`
+        )} checks`
+      );
+    } catch (e) {
+      this.totalFailedTests += test.expectedEntities.length;
+      logger.warn(`Test: ${test.name} field due to runtime error`, e);
+      this.failedTestsSummary.push({
+        testName: test.name,
+        entityId: undefined,
+        entityName: undefined,
+        failedAttributes: [`Runtime Error:\n${e.stack}`],
+      });
```

I think the runtime error should be specific to indexBlock like it was before. 

We should instead throw an error here that the test suite failed